### DISCUSSION TITLE
How to identify the selected app and it should open the selected app on mobile.

### DISCUSSION THREAD

**Discussion by kumarnagender09** [WalletConnect/walletconnect-monorepo#947] [Developer Support]
<img width="460" alt="Screenshot 2022-04-11 at 4 08 00 PM" src="https://user-images.githubusercontent.com/1977940/163335955-de9cc051-1e63-4327-bc08-3e30cfd5b25e.png">
![IMG_8983](https://user-images.githubusercontent.com/1977940/163335988-296b4b7e-01b5-45e2-99e6-d3c12ab8e2bc.PNG)
Hi @chadyj ,
How to open the selected wallet  using wallet connect 

Wallet connect response : 

wc:465680CC-541A-41AA-88BA-4B05D7C44CDA@1?bridge=https%3A%2F%2Fsafe-walletconnect.gnosis.io%2F&key=1d68bdd9f60dbe63a4f1d985e1cb8ca4797c73563def80fe747e07c63a96e2c9

in my ios swift code:
if let url = URL(string: connectionUrl), UIApplication.shared.canOpenURL(url){
            UIApplication.shared.open(url, options: [:], completionHandler: nil)
}

How to know about which app will open and can i do here to open the selected wallet so that it will open the selected wallet like rainbow or metamask or.....whatever i choose

**Comment by kumarnagender09** [comment]
@chadyj  can you please help me here

**Comment by kumarnagender09** [comment]
I want to perform like this
https://www.dropbox.com/s/3fe7qk4aqxyteeu/Video%20Apr%2005%2C%2016%2032%2005.mp4?dl=0

**Reply by chadyj** [reply]
You can create your own UI using our registry API with wallet information
https://docs.walletconnect.com/2.0/api/registry-api

**Reply by kumarnagender09** [reply]
@chadyj  I tried the same thing and used the universal links to open the wallet app from my own app. it is opening the selected wallet app but it does not establish the connection with wallet apps like show connection popup like cancel or connect.

App Name : Rainbow
universal links :  https://rnbwapp.com/wc?uri=wc:60C79615-EAA3-42B6-A4FE-AE4342B4004B@1?bridge=https%3A%2F%2Fsafe-walletconnect.gnosis.io%2F&key=81ef0dd124ae7045292a25bf5bddada26662a1b12e0c5ab50b29b8af4c5c6e13

App Name : MetaMask
universal links :  https://metamask.app.link/wc?uri=wc:60C79615-EAA3-42B6-A4FE-AE4342B4004B@1?bridge=https%3A%2F%2Fsafe-walletconnect.gnosis.io%2F&key=81ef0dd124ae7045292a25bf5bddada26662a1b12e0c5ab50b29b8af4c5c6e13

App Name : Argent
universal links :  https://argent.link/app/wc?uri=wc:60C79615-EAA3-42B6-A4FE-AE4342B4004B@1?bridge=https%3A%2F%2Fsafe-walletconnect.gnosis.io%2F&key=81ef0dd124ae7045292a25bf5bddada26662a1b12e0c5ab50b29b8af4c5c6e13

App Name : Trust Wallet
universal links :  https://link.trustwallet.com/wc?uri=wc:60C79615-EAA3-42B6-A4FE-AE4342B4004B@1?bridge=https%3A%2F%2Fsafe-walletconnect.gnosis.io%2F&key=81ef0dd124ae7045292a25bf5bddada26662a1b12e0c5ab50b29b8af4c5c6e13

**Reply by kumarnagender09** [reply]
Example to create the  uni link //    https://example.wallet/wc?uri=wc:00e46b69-d0cc-4b3e-b6a2-cee442f97188@1?bridge=https%3A%2F%2Fbridge.walletconnect.org&key=91303dedf64285cbbaf9120f6e9d160a5c8aa3deb67017a3874cd272323f48ae

    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        print("App Name : \(arWallets[indexPath.row].name)")
       
        var urlString = arWallets[indexPath.row].ref
        urlString.append("/wc?uri="+connectionUrl)
        print("universal links : ", urlString)
        if let url = URL(string: urlString), UIApplication.shared.canOpenURL(url){
            UIApplication.shared.open(url, options: [:], completionHandler: nil)
        }
    }

**Comment by kumarnagender09** [comment]
This is opening the app but not showing the connection popup in MetaMask and this is happening with other wallets also

https://metamask.app.link/wc?uri=wc:10876E8F-BBFB-44C8-A190-40E389B051DC@1?bridge=https%3A%2F%2Fbridge.walletconnect.org&key=de91fa60db4fbe64bafb2a4deba06791f1876a9c340d2f65159701f22d012313

@chadyj  @janus  @julien51 @bpierre  Please help me what is wrong with that

**Comment by frankorbit** [comment]
Hello, do you know how to get  the wallet apps list on the phone?
